### PR TITLE
Full draft of Rust parser

### PR DESCRIPTION
**Pull request by affo**
### Done checklist
- [x] `peek_with_regex` -> the Rust scanner doesn't offer `scan_with_regex` yet
- [x] more test cases
- [ ] source location

### CODE REVIEW

File: internal/rust/parser/src/strconv.rs [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+        None => return Err("invalid escape sequence".to_string()),
+    };
+    Ok(())
+}
+
+fn to_byte(c: char) -> Option<u8> {
+    match c {
+        '0'...'9' => Some(c as u8 - b'0'),
+        'a'...'f' => Some(c as u8 - b'a' + 10),
+        'A'...'F' => Some(c as u8 - b'A' + 10),
```
Comment (nathanielc) [inline internal/rust/parser/src/strconv.rs]: My compiler gave me a warning about this syntax being deprecated and it suggests something like `..=`

Comment (nathanielc) [reply internal/rust/parser/src/strconv.rs]: here is the complete compile warning:

```
warning: `...` range patterns are deprecated
  --> src/strconv.rs:76:12
   |
76 |         'A'...'F' => Some(c as u8 - b'A' + 10),
   |            ^^^ help: use `..=` for an inclusive range

```

Comment (affo) [reply internal/rust/parser/src/strconv.rs]: I don't get those warning event after `rustup update`.  
Do you get those in `cargo build`?

File: internal/rust/parser/src/lib.rs [hunk 2]
```diff
@@ -302,308 +541,805 @@ impl Parser {
...
+            self.consume();
+        }
+        props.append(&mut self.parse_property_list());
+        props
+    }
+    fn parse_property_list(&mut self) -> Vec<Property> {
+        let mut params = Vec::new();
+        let mut errs = Vec::new();
+        while self.more() {
+            let mut p: Property;
```
[the remark this answers is not in the document]
Comment (affo) [reply internal/rust/parser/src/lib.rs]: I don't get those warning event after `rustup update`.  
Do you get those in `cargo build`?

[the remark this answers is not in the document]
Comment (affo) [reply internal/rust/parser/src/lib.rs]: Anyways I updated and everything works... Maybe I'm running the wrong command when building.

File: internal/rust/parser/src/strconv.rs (hunk 1 above)
Comment (affo) [reply internal/rust/parser/src/strconv.rs]: Anyways I updated and everything works... Maybe I'm running the wrong command when building.

File: internal/rust/parser/src/lib.rs [hunk 3]
```diff
@@ -302,308 +541,805 @@ impl Parser {
...
+        let mut expr = self.parse_primary_expression();
+        loop {
+            // TODO(affo): find alternative way without cloning (borrowing issues).
+            let po = self.parse_postfix_operator(expr.clone());
+            match po {
+                Some(e) => expr = e,
+                None => break,
+            }
+        }
+        expr
```
[the remark this answers is not in the document]
Comment (affo) [reply internal/rust/parser/src/lib.rs]: That is elegant and cool. Anyways I got a hard time to understand what was going on 😅 .

I found a simpler solution that I just committed, what do you think?

[the remark this answers is not in the document]
Comment (nathanielc) [reply internal/rust/parser/src/lib.rs]: I like it, much simpler and very clear what is going on.

File: internal/rust/parser/src/strconv.rs (hunk 1 above)
Comment (nathanielc) [reply internal/rust/parser/src/strconv.rs]: Maybe do `cargo clean` and then `cargo build`? I noticed that the warnings only show up when it actually re compiles those bits, which isn't often.

Comment (affo) [reply internal/rust/parser/src/strconv.rs]: Yeah, I tried `clean` + `build` but I only got warnings on the unused C variables 😓

**Pull request merged**